ostd/src/mm/heap_allocator/mod.rs

```diff
@@ -54,25 +55,50 @@ impl LockedHeapWithRescue {
         self.heap
             .call_once(|| SpinLock::new(Heap::new(start as usize, size)));
     }
+
+    /// SAFETY: The range [start, start + size) must be a valid memory region.
+    unsafe fn add_to_heap(&self, start: usize, size: usize) {
+        self.heap
+            .get()
+            .unwrap()
+            .disable_irq()
+            .lock()
+            .add_memory(start, size);
+    }
+
+    fn avoid_low_memory(&self, remain_bytes: usize, layout: Layout) {
+        if remain_bytes <= PAGE_SIZE * 4 {
```

Comment:
The function name is inaccurate because when `remain_bytes <= PAGE_SIZE * 4`, the available memory is already low. So this function cannot _avoid_ it. I think calling it `rescue_low_memory` is fine.

Reply:
Sorry, second thought. The last name proposed by me is not good because it sounds exactly like `rescue`. I suggest renaming it `rescue_if_low_memory`.